docs/contributing/running-and-writing-acceptance-tests.md

```diff
@@ -1136,47 +1162,248 @@ The below are style-based items that _may_ be noted during review and are recomm
+### Avoiding Hard Coding
+Avoid hard coding key values in acceptance tests for consistency and testing flexibility. Resource testing is expected to pass across multiple AWS environments supported by the Terraform AWS Provider (e.g. AWS Standard and AWS GovCloud (US)). Contributors are not expected or required to perform testing outside of AWS Standard, e.g. running only in the `us-west-2` region is perfectly acceptable. However, contributors are expected to avoid hard coding with these guidelines.
 
-  ```hcl
-  data "aws_ami" "amzn-ami-minimal-hvm-ebs" {
-    most_recent = true
-    owners      = ["amazon"]
+#### Hardcoded Account IDs
 
-    filter {
-      name = "name"
-      values = ["amzn-ami-minimal-hvm-*"]
-    }
-    filter {
-      name = "root-device-type"
-      values = ["ebs"]
-    }
-  }
-  ```
+- [ ] __Uses Account Data Sources__: Any hardcoded account numbers in configuration, e.g., `137112412989`, should be replaced with a data source. Depending on the situation, there are several data sources for account IDs including:
+    - [`aws_caller_identity` data source](https://www.terraform.io/docs/providers/aws/d/caller_identity.html),
+    - [`aws_canonical_user_id` data source](https://www.terraform.io/docs/providers/aws/d/canonical_user_id.html),
+    - [`aws_billing_service_account` data source](https://www.terraform.io/docs/providers/aws/d/billing_service_account.html), and
+    - [`aws_sagemaker_prebuilt_ecr_image` data source](https://www.terraform.io/docs/providers/aws/d/sagemaker_prebuilt_ecr_image.html).
+- [ ] __Uses testAccCheckResourceAttrAccountID()__: Tests should utilize the available AWS Account ID check function, `testAccCheckResourceAttrAccountID()` to validate account ID attribute values in the Terraform state over `resource.TestCheckResourceAttrSet()` and `resource.TestMatchResourceAttr()`
+
+Here's an example of using `aws_caller_identity`:
+
+```hcl
+data "aws_caller_identity" "current" {}
+
+resource "aws_backup_selection" "test" {
+  plan_id      = aws_backup_plan.test.id
+  name         = "tf_acc_test_backup_selection_%[1]d"
+  iam_role_arn = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/service-role/AWSBackupDefaultServiceRole"
+}
+```
+
+#### Hardcoded AMI IDs
+
+- [ ] __Uses aws_ami Data Source__: Any hardcoded AMI ID configuration, e.g. `ami-12345678`, should be replaced with the [`aws_ami` data source](https://www.terraform.io/docs/providers/aws/d/ami.html) pointing to an Amazon Linux image. Use the convenience function called `testAccLatestAmazonLinuxHvmEbsAmiConfig()` (defined in `resource_aws_instance_test.go`) to declare `data "aws_ami" "amzn-ami-minimal-hvm-ebs" {...}`. Other convenience functions are also available including `testAccLatestAmazonLinuxHvmInstanceStoreAmiConfig()`, `testAccLatestAmazonLinuxPvEbsAmiConfig()`, `testAccLatestAmazonLinuxPvInstanceStoreAmiConfig`, and `testAccLatestWindowsServer2016CoreAmiConfig()`.
```

Comment:
I would recommend listing out the configuration helper functions and not calling out the physical location of functions which will change over time:

```suggestion
- [ ] __Uses aws_ami Data Source__: Any hardcoded AMI ID configuration, e.g. `ami-12345678`, should be replaced with the [`aws_ami` data source](https://www.terraform.io/docs/providers/aws/d/ami.html) pointing to an Amazon Linux image. The codebase includes test configuration helper functions to simplify these lookups:
    - `testAccLatestAmazonLinuxHvmEbsAmiConfig()`: The most recommended AMI lookup using Amazon Linux, HVM virtualization, and EBS storage. To reference the AMI ID in the test configuration: `data.aws_ami.amzn-ami-minimal-hvm-ebs.id`.
    - `testAccLatestAmazonLinuxHvmInstanceStoreAmiConfig()`: AMI lookup using Amazon Linux, HVM virtualization, and Instance Store storage. Should only be used in testing that requires that type of storage over EBS. To reference the AMI ID in the test configuration: `data.aws_ami.amzn-ami-minimal-hvm-instance-store.id`.
    - `testAccLatestAmazonLinuxPvEbsAmiConfig()`: AMI lookup using Amazon Linux, Paravirtual virtualization, and EBS storage. Should only be used in testing that requires Paravirtual virtualization over HVM. To reference the AMI ID in the test configuration: `data.aws_ami.amzn-ami-minimal-pv-ebs.id`.
    - `testAccLatestAmazonLinuxPvInstanceStoreAmiConfig`: AMI lookup using Amazon Linux, Paravirtual virtualization, and Instance Store storage. Should only be used in testing that requires Paravirtual virtualization over HVM and Instance Store over EBS. To reference the AMI ID in the test configuration: `data.aws_ami.amzn-ami-minimal-pv-instance-store.id`.
    - `testAccLatestWindowsServer2016CoreAmiConfig()`: AMI lookup using Windows Server 2016 Core, HVM virtualization, and EBS storage. Should only be used in testing that requires Windows. To reference the AMI ID in the test configuration: `data.aws_ami.win2016core-ami.id`.
```